pkg/operation/botanist/dns.go

```diff
@@ -40,22 +44,23 @@ const (
 	DNSPurposeInternal = "internal"
 	// DNSPurposeExternal is a constant for a DNS record used for the external domain name.
 	DNSPurposeExternal = "external"
+	// DNSProviderRolePrimary is a constant for a DNS providers used to manage shoot domains.
+	DNSProviderRolePrimary = "primary-dns-provider"
+	// DNSProviderRoleAdditional is a constant for additionally managed DNS providers.
+	DNSProviderRoleAdditional = "managed-dns-provider"
 )
 
 // DeployInternalDomainDNSRecord deploys the DNS record for the internal cluster domain.
 func (b *Botanist) DeployInternalDomainDNSRecord(ctx context.Context) error {
-	if err := b.deployDNSProvider(ctx, DNSPurposeInternal, b.Garden.InternalDomain.Provider, b.Garden.InternalDomain.SecretData, []string{b.Shoot.InternalClusterDomain}, nil, b.Garden.InternalDomain.IncludeZones, b.Garden.InternalDomain.ExcludeZones); err != nil {
+	if err := b.deployDNSProvider(ctx, DNSPurposeInternal, DNSProviderRolePrimary, b.Garden.InternalDomain.Provider, b.Garden.InternalDomain.SecretData, []string{b.Shoot.InternalClusterDomain}, nil, b.Garden.InternalDomain.IncludeZones, b.Garden.InternalDomain.ExcludeZones); err != nil {
```

Comment:
I know its not part of this PR, but passing purpose to a name field looks a bit confusing. Could we do something link 

```go
const DNSInternalProviderName = DNSPurposeInternal 
``` 

and use that instead?